db/db_impl_open.cc

```diff
@@ -356,6 +356,16 @@ Status DBImpl::Recover(
       assert(s.IsIOError());
       return s;
     }
+    // Verify compatibility of env_options_ and filesystem
+    {
+      unique_ptr<SequentialFile> idfile;
+      s = env_->NewSequentialFile(IdentityFileName(dbname_), &idfile,
+                                  env_options_);
+      if (!s.ok()) {
+        return Status::NotSupported(
+            "Found options incompatible with filesystem. Check your options.");
```

Comment:
Need to pass the information of s into this error message.
Also Status::NotSupported() means RocksDB doesn't support it. If users set up something wrong by themselves, Status::InvalidArgument() is more appropriate.

Reply:
Not sure what's the best way to help use debug the incorrect argument. With `s.getState()` fix, the current message is:
> open error: Invalid argument: Found options incompatible with filesystem: While open a file for random read: /dev/shm/dbbench/IDENTITY: Invalid argument
It's not very clear that -use_direct_reads is the problem here but I guess we can live with it?

Reply:
@miasantreble we don't know whether it's the problem of -use_direct_reads either, right?

Reply:
Another issue: env_options_ doesn't cover all the direct I/O case. We have two options: https://github.com/facebook/rocksdb/blob/master/include/rocksdb/options.h#L584-L601
and test test only covers only one of them. We should set the direct I/O options in a customized EnvOptions when either of the two is true.